### PR TITLE
Refactor SecurityAnalysisResult and LimitViolation classes

### PR DESCRIPTION
**Pull request by mathbagu**

### CODE REVIEW

File: action/action-simulator/src/main/java/com/powsybl/action/simulator/loadflow/LoadFlowActionSimulator.java [hunk 1]
```diff
@@ -148,20 +136,16 @@ private boolean next(ActionDb actionDb, RunningContext context) {
...
         }
         if (result.isOk()) {
-            List<LimitViolation> violations = LIMIT_VIOLATION_FILTER.apply(Security.checkLimits(context.getNetwork(), 1));
-            if (violations.size() > 0) {
-                LOGGER.info("Violations: \n{}", Security.printLimitsViolations(violations, NO_FILTER));
-            }
-            if (observers != null) {
-                observers.forEach(o -> o.loadFlowConverged(context.getContingency(), violations));
+            List<LimitViolation> violations = LIMIT_VIOLATION_FILTER.apply(Security.checkLimits(context.getNetwork(), 1), context.getNetwork());
+            if (violations.isEmpty()) {
```
Comment (geofjamg) [inline action/action-simulator/src/main/java/com/powsybl/action/simulator/loadflow/LoadFlowActionSimulator.java]: !violation.isEmpty()

File: security-analysis/src/main/java/com/powsybl/security/Security.java [hunk 2]
```diff
@@ -255,32 +234,174 @@ public static void printPreContingencyViolations(SecurityAnalysisResult result,
...
+                    .writeCell(getViolationLimit(violation), HorizontalAlignment.RIGHT, numberFormatter)
+                    .writeCell(getViolation(violation), HorizontalAlignment.RIGHT, numberFormatter)
+                    .writeCell(getLoadingRate(violation), HorizontalAlignment.RIGHT, numberFormatter);
             } catch (IOException e) {
                 throw new UncheckedIOException(e);
             }
         };
     }
 
+    private static String getCountry(LimitViolation limitViolation, Network network) {
```
Comment (geofjamg) [inline security-analysis/src/main/java/com/powsybl/security/Security.java]: getCountryString ?

File: security-analysis/src/main/java/com/powsybl/security/Security.java [hunk 3]
```diff
@@ -255,32 +234,174 @@ public static void printPreContingencyViolations(SecurityAnalysisResult result,
...
+        Country country = (network == null) ? null : LimitViolation.getCountry(limitViolation, network);
+
+        return Objects.toString(country, "");
+    }
+
+    private static String getNominalVoltage(LimitViolation limitViolation, Network network) {
```
Comment (geofjamg) [inline security-analysis/src/main/java/com/powsybl/security/Security.java]: getNominalVoltageString ?

File: security-analysis/src/main/java/com/powsybl/security/SecurityAnalysisResult.java [hunk 4]
```diff
@@ -14,16 +16,31 @@
  */
 public class SecurityAnalysisResult {
 
+    private final Network network;
```
Comment (geofjamg) [inline security-analysis/src/main/java/com/powsybl/security/SecurityAnalysisResult.java]: Not a good idea to put network here as you wont be able after a json loop to recreate it.

Comment (mathbagu) [reply security-analysis/src/main/java/com/powsybl/security/SecurityAnalysisResult.java]: The network is skipped by the exporter. I’ll try to find a better way to handle the network

**Pull request closed without merging**